libnetwork/drivers/overlay/ovmanager/ovmanager.go

```diff
@@ -89,14 +89,10 @@ func (d *driver) NetworkAllocate(id string, option map[string]string, ipV4Data,
 	for key, val := range option {
 		if key == netlabel.OverlayVxlanIDList {
 			log.G(context.TODO()).Debugf("overlay network option: %s", val)
-			valStrList := strings.Split(val, ",")
-			for _, idStr := range valStrList {
-				vni, err := strconv.Atoi(idStr)
-				if err != nil {
-					return nil, fmt.Errorf("invalid vxlan id value %q passed", idStr)
-				}
-
-				vxlanIDList = append(vxlanIDList, uint32(vni))
+			var err error
+			vxlanIDList, err = overlayutils.AppendVNIList(vxlanIDList, val)
```

Comment:
I was thinking more along the lines of only parsing each value as needed down at 109.

Reply:
This PR is already a premature optimization for a code path which will see one, maybe two elements for any practical real-world usage. This is as far as I'm willing to go.

Reply:
For me it is not so much optimization as it is safety.
But I see your point. I don't really see this particular change as an improvement for performance or safety.